### ISSUE TITLE
calcite-input suffix-text is not announced by screen-reader

### ISSUE THREAD

**Issue by tom12993** [Esri/calcite-components#14718]
**monday.com sync:** #12403051300

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Input Number

### Summary

The accessibility team flagged this as a Severity-8 violation on our app.

### Actual Behavior

When focusing on the `calcite-number-input` with `suffix-text`, the screen-reader only reads the number value, not the suffix.

### Expected Behavior

If an Input Number has suffix text, it should be part of the value when announced by the screen-reader.
For example when focusing on the following, the screen-reader should announce "edit  selected 123 users", not just "edit  selected 123"
```
<calcite-input-number value="123" suffix-text="users"></calcite-input-number>
```

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/input-number/

### Reproduction Steps

1. On `calcite-input-number` component, set `suffix-text` property to "users" and `value` property to "123"
2. Tab into the component in the DOM
3. The screen-reader will only announce the value, but not the suffix-text

### Reproduction Version

v5.1

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Monitor

**Comment by geospatialem** [comment]
The above should be handled with an accompanied visual label so more individuals are aware of the units specified by a prefix or suffix. There is no native support available, but other design systems also follow these patterns. For instance:
- [U.S. Web Design System](https://designsystem.digital.gov/components/input-prefix-suffix/)
- [GOV.UK Design system](https://design-system.service.gov.uk/components/text-input/#:~:text=Asking%20for%20decimal%20numbers)

Recommended implementation with `label-text` or `calcite-label`:
```html
<!-- Using label-text -->
<calcite-input-number label-text="Registered users" suffix-text="users" min="0" placeholder="1" step="1"></calcite-input-number>

<!-- Using Label for custom formatting -->
<calcite-label layout="inline">Registered users
    <calcite-input-number suffix-text="users" min="0" placeholder="1" step="1"></calcite-input-number>
</calcite-label>
```

<img width="531" height="122" alt="Image" src="https://github.com/user-attachments/assets/1972ffc9-8ac5-4501-a1cf-a81050b2fac0" />

Updating the above as a documentation update to include a sample for the `input-number` component, since it was recently changed (go live is tomorrow). 

cc @Esri/calcite-pes

**Comment by tom12993** [comment]
@geospatialem Thank you for providing the recommended approach!
